src/mpid/ch4/netmod/ofi/globals.c

```diff
@@ -148,6 +148,31 @@ MPIDI_OFI_capabilities_t MPIDI_OFI_caps_list[MPIDI_OFI_NUM_SETS] =
      .major_version = MPIDI_OFI_MAJOR_VERSION_BGQ,
      .minor_version = MPIDI_OFI_MINOR_VERSION_BGQ}
     ,
+    {   /* cxi */
+     .enable_av_table = MPIDI_OFI_ENABLE_AV_TABLE_CXI,
+     .enable_scalable_endpoints = MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS_CXI,
+     .enable_shared_contexts = MPIDI_OFI_ENABLE_SHARED_CONTEXTS_CXI,
+     .enable_mr_virt_address = MPIDI_OFI_ENABLE_MR_VIRT_ADDRESS_CXI,
+     .enable_mr_allocated = MPIDI_OFI_ENABLE_MR_ALLOCATED_CXI,
+     .enable_mr_prov_key = MPIDI_OFI_ENABLE_MR_PROV_KEY_CXI,
+     .enable_tagged = MPIDI_OFI_ENABLE_TAGGED_CXI,
+     .enable_am = MPIDI_OFI_ENABLE_AM_CXI,
+     .enable_rma = MPIDI_OFI_ENABLE_RMA_CXI,
+     .enable_atomics = MPIDI_OFI_ENABLE_ATOMICS_CXI,
+     .enable_data_auto_progress = MPIDI_OFI_ENABLE_DATA_AUTO_PROGRESS_CXI,
+     .enable_control_auto_progress = MPIDI_OFI_ENABLE_CONTROL_AUTO_PROGRESS_CXI,
+     .enable_pt2pt_nopack = MPIDI_OFI_ENABLE_PT2PT_NOPACK_CXI,
+     .num_am_buffers = MPIDI_OFI_NUM_AM_BUFFERS_CXI,
+     .max_endpoints = MPIDI_OFI_MAX_ENDPOINTS_CXI,
+     .max_endpoints_bits = MPIDI_OFI_MAX_ENDPOINTS_BITS_CXI,
+     .fetch_atomic_iovecs = MPIDI_OFI_FETCH_ATOMIC_IOVECS_CXI,
+     .context_bits = MPIDI_OFI_CONTEXT_BITS_CXI,
+     .source_bits = MPIDI_OFI_SOURCE_BITS_CXI,
+     .tag_bits = MPIDI_OFI_TAG_BITS_CXI,
+     .counter_wait_objects = MPIDI_OFI_COUNTER_WAIT_OBJECTS_CXI,
+     .major_version = MPIDI_OFI_MAJOR_VERSION_CXI,
+     .minor_version = MPIDI_OFI_MINOR_VERSION_CXI}
+    ,
```

Comment:
The `globals.c` file is a bad example because all of these structs are the same, but with different provider names at the end (opportunity for future code generation 😄?). This is here to assist in runtime provider selection. I think this comment belongs in `ofi_capability_sets.h`, where the answer would be that the capability set for CXI is different from the default set:

* CXI uses only regular endpoints (for now)
* CXI does not support shared contexts
* CXI has a smaller tag space and only supports 20 bits of tag space (for now)
* CXI will support optimized memory regions once we figure out the issues integrating support for RMA.

Reply:
Thanks. Can we add these as comments in `ofi_capability_sets.h`? I am thinking that we should have these summary comments for all the static provider set.

Reply:
I'm not sure it makes sense to put all of these in as comments as some of them are pretty self explanatory (e.g., shared contexts), but I'll add a few and you can take a look.

Reply:
It's not about why we need the settings, but more about provide some summary to understand why we need maintain a separate capability set rather than reduce to default and fine-tune the settings at init-time. Reading the capability set one by one quickly numbs my brain 😃

Reply:
You're asking why capability sets exist in general?

Reply:
Yeah, we don't need high light the shared context unless the runtime capability won't work for it.

Reply:
> 

Yeah, something like that. For example, we don't maintain capability set for every provider. The static capability set, while works, it is essentially a hack, it makes our implementation less able to adapt to changes in the upstream libfabric.